htdocs/Frameworks/moduleclasses/moduleadmin/moduleadmin.php

```
@@ -506,6 +506,36 @@ public function renderInfoBox()
+     *
+     * @return string sanitized HTML fragment
+     */
+    public static function sanitizeChangelogLine($line)
+    {
+        static $allowed = 'h1|h2|h3|h4|h5|h6|p|br|hr|ul|ol|li|strong|b|em|i|u|code|pre|blockquote|span';
+
+        $escaped = htmlspecialchars((string) $line, ENT_QUOTES | ENT_SUBSTITUTE, 'UTF-8');
+
+        return preg_replace_callback(
+            '#&lt;(/?)\s*(' . $allowed . ')\s*(/?)&gt;#i',
+            static fn($m) => '<' . $m[1] . strtolower($m[2]) . $m[3] . '>',
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $m. Configured minimum length is 3. (undefined)

(ShortVariable)